Source/WebCore/rendering/RenderLayerScrollableArea.cpp

```diff
@@ -845,14 +845,22 @@ bool RenderLayerScrollableArea::canShowNonOverlayScrollbars() const
     return canHaveScrollbars() && !(m_layer.renderBox() && m_layer.renderBox()->canUseOverlayScrollbars());
 }
 
-void RenderLayerScrollableArea::createScrollbarsController()
+void RenderLayerScrollableArea::updateScrollbarsControllerForLayerCreation()
 {
-    if (usesAsyncScrolling()) {
+    if (!hasRemoteScrollbarsController()) {
         if (auto scrollbarController = m_layer.page().chrome().client().createScrollbarsController(m_layer.page(), *this)) {
```

Comment:
Why not just call `createScrollbarsController()` here?

Testing `! hasRemoteScrollbarsController()` then calling `chrome().client().createScrollbarsController()` implies some knowledge that it's the client that creates the `RemoteScrollbarsController`, but this code shouldn't have to know about that.

Instead this should do something like ask the client "does this RenderLayerScrollableArea have the right type of scrollbars controller already?"

Reply:
Right yeah this should be the other way around.